src/daemon/cJsonParser.cc

```diff
@@ -65,9 +56,35 @@ bool cJsonParser::getConfig(struct sJsonDevicesConfig* pConfig)
         g_object_unref(pReader);
         return false; // failure
     }
+
+    for (int i = 0; i < elementCount; ++i)
+    {
+        json_reader_read_element(pReader, i);
+        pError = (GError*)json_reader_get_error(pReader);
+        if (pError)
+        {
+            LOG_EVENT(LOG_ERR, "Error parsing 'devices': %s\n",
+                pError->message);
+            json_reader_end_member(pReader);
+            g_object_unref(pReader);
+            return false; // failure
+        }
+
+        auto devicePath = json_reader_get_string_value(pReader);
+        if (devicePath != nullptr)
+            pConfig->devices.emplace_back((std::string)devicePath);
+        json_reader_end_element(pReader);
+    }
+    return true; // success
+}
+
+bool cJsonParser::getArrayPostV18(JsonReader* pReader, struct sJsonDevicesConfig* pConfig)
```

Comment:
This needs to be inside an `#if (JSON_CHECK_VERSION(1, 8, 0))` too

Reply:
should I just have one `getArray` function and wrap that inside the `#if (JSON_CHECK_VERSION(1, 8, 0))`?

Reply:
I think it's Pawel who cares more about coding style. From my perspective everything that relies on >=v1.8 has to be #ifdef'ed out in a <v1.8 build

Reply:
I guess yeah having just one function that is #ifdefed is preferable otherwise the code will be like a swiss cheese it seems. So my latest opinion is that to be consistent: #ifdef whole getArray and getConfig and just be done with it.

Reply:
I suppose the switch could be in CMAKE itself which would be the cleanest solution, which I think I have seen somewhere.